prereise/gather/demanddata/transportation_electrification/mileage.py

```diff
@@ -0,0 +1,201 @@
+
+    # Somehow load in the matfile
+    nhts_census = loadmat(filepath)
+    # Load the data for the desired region
+    raw_data = nhts_census[f"census_{census_region}_sorted"]
+    # Convert to data frame, adding column names
+    df = pd.DataFrame(raw_data, columns=const.nhts_census_column_names)
+
+    return df
+
+
+def remove_ldt(data: pd.DataFrame) -> pd.DataFrame:
+    """Remove light duty trucks from data loaded from nths_census.mat.
+
+    :param pandas.DataFrame data: the data returned from load_data.
+    :return: (*pandas.DataFrame*) -- the data loaded from load_data with all rows
+        involving LDT removed.
+    """
+
+    return data.loc[data["Vehicle type"] < 5].copy()
+
+
+# ldt = light duty truck
+# ldv = light duty vehicle
+def total_daily_vmt(
+    census_region: int,
+    comm_type: int,
+    locationstrategy: list[int],
+    input_day: list[int],
+) -> np.array[list[float]]:
+    """load data and use the parameters to calculate total_daily_vmt.
+
+    :param int census_region: the type of census
+    :param int comm_type: the type of Commute
+    :param list locationstrategy: strategy for each location
+    :param list input_day: day of the week for each day in the year derived from
+        first_func
+    :return: (*list*) -- daily_vmt_total each row is a year of entries for each vehicle
+        type
+    """
+    # get the data
+    data = remove_ldt(load_data(census_region))
+    n = len(data)
+
+    if comm_type == 1:
+        # removes VMT for trips from work->home and home-> work
+        # (not exactly correct due to chained trips involving work and home but no
+        # way to remove the data)
+        if all([i != 2 for i in locationstrategy]):
+            locationstrategy = 2
+            # putting this part in function means rest of code won't be able to see
+            # change in locationstrategy value.
+            # will have to use global or return locationstrategy and reassign
+            print('"locationstrategy" changed to "Home and Work" for comm_type == 1')
+        # isolates home->work trips
+        trip_home_work = data.loc[data["why from"] == 1 and data["why to"] == 11]
+        # isolates work->home trips
+        trip_work_home = data.loc[data["why from"] == 11 and data["why to"] == 1]
+
+        # set trips from work to home to 0 distance
+        data.loc[trip_work_home.index, "Miles traveled"] = 0
+        data.loc[trip_home_work.index, "Miles traveled"] = 0
+
+    # pre-processing of vehicle trip data to determine weekday or weekend
+    # weekend in the case of 1 or 7
+    data.loc[data["Day of Week"].isin({1, 7}), "If Weekend"] = 1
+    # weekday in the case of 2-6 (inclusive)
+    data.loc[data["Day of Week"].isin(list(range(2, 7))), "If Weekend"] = 2
+
+    day2 = get_day2(data)
+    daily_vmt_total = np.array([[0, 0] for i in range(365)])
+
+    for day_iter in range(365):
+        for i in range(n):
+            if day2[i] == input_day[day_iter]:

```

Comment:
If `day2` is a vector of weekday-or-weekend, then why do we compare this with the day of the week in `input_day`?

Reply:
This is where GroupBy would come in as you mentioned in Slack - in the refactor these both are IfWeekend values